lib/jekyll-github-metadata/ghp_metadata_generator.rb

```diff
@@ -5,18 +5,31 @@ module GitHubMetadata
     class GHPMetadataGenerator < Jekyll::Generator
       safe true
 
+      attr_reader :site
+
       def generate(site)
         Jekyll::GitHubMetadata.log :debug, "Initializing..."
+        @site = site
+        site.config["github"]  = github_namespace
+        site.config["url"]     = drop.url if site.config["url"].nil?
```

Comment:
This could be `site.config["url"] ||= drop.url` and same for `baseurl`. Would save you an explicit `if` 😄